lib/services/file/fileservice.core.js

```diff
@@ -917,9 +981,15 @@ FileService.prototype.deleteShare = function (share, optionsOrCallback, callback
   });
 
   var options = extend(true, {}, userOptions);
+  if (!azureutil.objectIsNull(options.shareSnapshotId) && !azureutil.objectIsNull(options.deleteSnapshots)) {
+    throw new ArgumentError('options', SR.INVALID_DELETE_SNAPSHOT_OPTION);
+  }
+
   var webResource = WebResource.del(share)
-    .withQueryOption(QueryStringConstants.RESTYPE, 'share');
-  
+    .withQueryOption(QueryStringConstants.RESTYPE, 'share')
+    .withQueryOption(QueryStringConstants.SHARE_SNAPSHOT, options.shareSnapshotId)
+    .withHeader(HeaderConstants.DELETE_SNAPSHOT, options.deleteSnapshots);
```

Comment:
This would be an object, is this wished?

Reply:
The query options and headers are all string. Which object do you refer to?